api/v1alpha4/machinedeployment_webhook.go

```diff
@@ -92,6 +93,34 @@ func (m *MachineDeployment) validate(old *MachineDeployment) error {
 		)
 	}
 
+	if m.Spec.Strategy != nil &&
```

Comment:
Should we add a unit test for this validations rules?

Reply:
done!